Sources/TestingInternals/Discovery.cpp

```diff
@@ -227,15 +234,32 @@ static SWTMachHeaderList getMachHeaders(void) {
   static constinit SWTMachHeaderList *machHeaders = nullptr;
   static constinit os_unfair_lock lock = OS_UNFAIR_LOCK_INIT;
 
+  // The bounds of the dyld shared cache. On platforms that support it (Darwin),
+  // most system images are containined in this range. System images can be
+  // expected not to contain test declarations, so we don't need to walk them.
+  static constinit uintptr_t dyldSharedCacheBegin = 0;
+  static constinit uintptr_t dyldSharedCacheEnd = 0;
+
   static constinit dispatch_once_t once = 0;
   dispatch_once_f(&once, nullptr, [] (void *) {
     machHeaders = reinterpret_cast<SWTMachHeaderList *>(std::malloc(sizeof(SWTMachHeaderList)));
     ::new (machHeaders) SWTMachHeaderList();
     machHeaders->reserve(_dyld_image_count());
 
+    // Get the bounds of the shared cache.
+    size_t dyldSharedCacheLength = 0;
+    dyldSharedCacheBegin = reinterpret_cast<uintptr_t>(_dyld_get_shared_cache_range(&dyldSharedCacheLength));
+    dyldSharedCacheEnd = dyldSharedCacheBegin + dyldSharedCacheLength;
+
     objc_addLoadImageFunc([] (const mach_header *mh) {
       auto mhn = reinterpret_cast<SWTMachHeaderList::value_type>(mh);
 
+      // Ignore this Mach header if it is in the shared cache.
+      auto machHeaderAddress = reinterpret_cast<uintptr_t>(mhn);
+      if (machHeaderAddress >= dyldSharedCacheBegin && machHeaderAddress < dyldSharedCacheEnd) {
```

Comment:
Should that be `machHeaderAddress =< dyldSharedCacheEnd`?

Reply:
I also asked myself that. But nope, it's just the upper bound, non-inclusive. Compare `startIndex` and `endIndex` in Swift ranges. (It won't matter either way since a valid Mach header can't be 1 byte long.)